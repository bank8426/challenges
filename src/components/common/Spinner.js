import React from 'react'
import styled, { keyframes } from 'styled-components';
const spinKeyframe = keyframes`
  0% {
    -webkit-transform: rotate(0deg);
    transform: rotate(0deg);
  }
  100% {
    -webkit-transform: rotate(360deg);
    transform: rotate(360deg);
  }
`

const StyledSpinner = styled.div`
  margin: 30px auto;
  font-size: 10px;
  position: relative;
  text-indent: -9999em;
  border-radius: 50%;
  border: 1em solid rgba(27, 149, 242, 0.2);
  border-left-color: #1b95f2;
  width: 3em;
  height: 3em;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -webkit-animation: ${spinKeyframe} 1.1s infinite linear;
  animation: ${spinKeyframe} 1.1s infinite linear;
  overflow: hidden;
`;

const Spinner = () => {
  return (
    <StyledSpinner/>
  )
}

export default Spinner
