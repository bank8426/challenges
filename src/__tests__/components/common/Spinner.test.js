import React from 'react'
import Spinner from '../../../components/common/Spinner'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

it('it should display spinner', () => {
  const tree = renderer.create(<Spinner />).toJSON()
  expect(tree).toMatchSnapshot()
})
