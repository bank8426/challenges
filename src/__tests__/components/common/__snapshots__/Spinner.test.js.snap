// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should display spinner 1`] = `
.c0 {
  margin: 30px auto;
  font-size: 10px;
  position: relative;
  text-indent: -9999em;
  border-radius: 50%;
  border: 1em solid rgba(27,149,242,0.2);
  border-left-color: #1b95f2;
  width: 3em;
  height: 3em;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -webkit-animation: hmHMaM 1.1s infinite linear;
  -webkit-animation: hmHMaM 1.1s infinite linear;
  animation: hmHMaM 1.1s infinite linear;
  overflow: hidden;
}

<div
  className="c0"
/>
`;
